src/libtiled/orthogonalrenderer.cpp

```diff
@@ -188,12 +206,17 @@ void OrthogonalRenderer::drawGrid(QPainter *painter, const QRectF &rect,
     if (tileWidth <= 0 || tileHeight <= 0)
         return;
 
-    const int startX = qMax(0, (int) (rect.x() / tileWidth) * tileWidth);
-    const int startY = qMax(0, (int) (rect.y() / tileHeight) * tileHeight);
-    const int endX = qMin(qCeil(rect.right()),
-                          map()->width() * tileWidth + 1);
-    const int endY = qMin(qCeil(rect.bottom()),
-                          map()->height() * tileHeight + 1);
+    int startX = ((int) (rect.x() / tileWidth - 1)) * tileWidth;
```

Comment:
I guess the -1 is to make it work in case of negative coordinates, but it means an extra column of tiles will be rendered in positive coordinate space. A real fix would be to use `qFloor(rect.x() / tileWidth) * tileWidth`.